oauth2_http/java/com/google/auth/oauth2/ExternalAccountSupplierContext.java

```diff
@@ -0,0 +1,94 @@
+package com.google.auth.oauth2;
+
+import com.google.auth.oauth2.ExternalAccountCredentials.SubjectTokenTypes;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.io.Serializable;
+
+/**
+ * Context object to pass relevant variables from external account credentials to suppliers. This
+ * will be passed on any call made to {@link IdentityPoolSubjectTokenSupplier} or {@link
+ * AwsSecurityCredentialsSupplier}.
+ */
+public class ExternalAccountSupplierContext implements Serializable {
+
+  private static final long serialVersionUID = -7852130853542313494L;
+
+  private final String audience;
+  private final String subjectTokenType;
+
+  /** Internal constructor. See {@link ExternalAccountSupplierContext.Builder}. */
+  private ExternalAccountSupplierContext(Builder builder) {
+    this.audience = builder.audience;
+    this.subjectTokenType = builder.subjectTokenType;
+  }
+
+  /**
+   * Returns the credentials' expected audience.
+   *
+   * @return the requested audience. For example:
+   *     "//iam.googleapis.com/locations/global/workforcePools/$WORKFORCE_POOL_ID/providers/$PROVIDER_ID".
+   */
+  public String getAudience() {
+    return audience;
+  }
+
+  /**
+   * Returns the credentials' expected subject token type.
+   *
```

Comment:
Let's document valid values that can be returned here.

Reply:
There's not really a static list of "valid" values since the backend could change without a client update. Added to the java doc that it expects an STS oauth2 subject token type - Same as how we document the field for external account credentials.